lib/perl/Genome/Sys/LockProxy.pm

```diff
@@ -0,0 +1,106 @@
+package Genome::Sys::LockProxy;
+
+use strict;
+use warnings;
+
+use Genome::Sys::Lock qw();
+use Params::Validate qw(validate_with);
+use Scalar::Util qw(blessed);
+
+=item new()
+
+Keyword Arguments:
+
+=over
+
+=item resource
+
+C<resource> is a unique identifier for some resource.
+
+=item scope
+
+C<scope> is the scope to which the lock is bound.  See C<Genome::Sys::Lock::scopes()> for valid scopes.
+
+=cut
+
+sub new {
+    my $class = shift;
+    my $spec = Genome::Sys::Lock::LOCK_RESOURCE_PARAMS_SPEC();
+    $spec->{resource} = delete $spec->{resource_lock};
+    my %params = validate_with(
+        params => \@_,
+        spec => $spec,
+    );
+    return bless \%params, $class;
+}
+
+=item resource
+
+Accesor for the C<resource> field.
+
+=cut
+
+sub resource {
+    my $self = shift;
+    return $self->{resource};
+}
+
+=item scope
+
+Accesor for the C<scope> field.
+
+=cut
+
+sub scope {
+    my $self = shift;
+    return $self->{scope};
+}
+
+=item lock()
+
+C<lock()> is an instance method with optional parameters C<block_sleep>,
+C<max_try>, or C<wait_announce_interval>.  It behaves similar to
+C<Genome::Sys::Lock::lock_resource()> but on sucess C<lock()> returns an
+instance of C<Genome::Sys::LockProxy> that can later be unlocked with
+C<unlock()>.
+
+=cut
+
+sub lock {
+    my $self = shift;
+    my %params = validate_with(
```

Comment:
This is just a lightweight check for parameter names.  The real validation is still in [Genome::Sys::Lock::lock_resource][1].  The lightweight check is duplication though.  I think I'll split `Genome::Sys::Lock::LOCK_RESOURCE_PARAMS_SPEC` up so I can use it here.

[1]: https://github.com/nnutter/genome/blob/988981633940f9cd10844450659399325e107601/lib/perl/Genome/Sys/Lock.pm#L73